src/compiler/checker.ts

```diff
@@ -9982,6 +9982,26 @@ namespace ts {
             if (checkType === wildcardType || extendsType === wildcardType) {
                 return wildcardType;
             }
+            // Simplifications for types of the form `T extends U ? U : never` and `T extends U ? never : T`.
+            const originalCheckType = getActualTypeVariable(root.checkType);
+            const trueType = instantiateType(root.trueType, mapper);
+            const falseType = instantiateType(root.falseType, mapper);

```

Comment:
These instantiations were always deferred before. Now they're eager. I'm somewhat concerned this creates extra work that we don't need to do.

Reply:
I could defer them to only when simplified again, but then only types with a declared `never` would be simplifiable, whereas like this one brance can be instantiated to never, allowing a broader set of conditional types to be reduced.